devtools/server/actors/replay/connection.js

```diff
@@ -1189,6 +1195,79 @@ async function uploadAllSourcemapAssets({
   ]);
 }
 
+async function writeAllSourcemapAssets({
+  recordingId,
+  sourceMapURL,
+  sourceMapBaseURL,
+  targetContentHash,
+  targetURLHash,
+  targetMapURLHash,
+}, recording) {
+  const contentPrincipal = recording.browser.contentPrincipal;
+  const result = await fetchText(contentPrincipal, recordingId, sourceMapURL);
+  if (!result) {
+    return;
+  }
+  const id = String(Math.floor(Math.random() * 10000000000));
+  const name = `sourcemap-${id}`;
+  await writeToRecordingDirectory(`${name}.map`, result.text);
+  await writeToRecordingDirectory(`${name}.metadata.json`, JSON.stringify({
+    kind: "sourcemap",
+    name: `${name}.map`,
+    url: sourceMapURL,
+    baseUrl: sourceMapBaseURL,
+    targetContentHash,
+    targetURLHash,
+    targetMapURLHash,
+  }));
+
+  const { sources } =
+    collectUnresolvedSourceMapResources(result.text, sourceMapURL, sourceMapBaseURL);
+
+  for (const { offset, url } of sources) {
+    const result = await fetchText(contentPrincipal, recordingId, url);
+    if (!result) {
+      continue;
+    }
+    const sourceId = String(Math.floor(Math.random() * 10000000000));
+    const name = `original-source-${id}-${sourceId}`;
+    await writeToRecordingDirectory(name, result.text);
+    await writeToRecordingDirectory(`${name}.metadata.json`, JSON.stringify({
+      kind: "original-source",
+      name,
+      parentId: id,
+      parentOffset: offset,
```

Comment:
I'd probably lean toward having `value: result.text` or something so we don't have to read two separate files.